src/app/containers/ChartbeatAnalytics/index.jsx

```diff
@@ -12,27 +15,39 @@ import {
   getDomain,
   buildSections,
   getType,
+  getTitle,
 } from '../../lib/analyticsUtils/chartbeat';
 
-const ChartbeatAnalytics = () => {
-  const { service } = useContext(ServiceContext);
-  const { env, platform, pageType } = useContext(RequestContext);
+import { getReferrer } from '../../lib/analyticsUtils';
+
+const ChartbeatAnalytics = ({ data }) => {
+  const { service, brandName } = useContext(ServiceContext);
+  const { env, platform, pageType, previousPath, origin } = useContext(
+    RequestContext,
+  );
   const { enabled } = useToggle('chartbeatAnalytics');
 
   if (!enabled) {
     return null;
   }
-
+  const referrer = getReferrer(platform, origin, previousPath);
+  const title = getTitle(pageType, data, brandName);
   const domain = env !== 'live' ? getDomain('test') : getDomain(service);
   const sections = buildSections(service, pageType);
   const cookie = getSylphidCookie();
   const type = getType(pageType);
+  const hasCookie = !!cookie;
+  const hasReferrer = !!referrer;
```

Comment:
Is it worth the overhead of the extra props? Can't we just check if the `cookie`/`referrer` props are falsy at the level of each component?